godot-macros/src/class/data_models/func.rs

```diff
@@ -252,22 +257,49 @@ fn make_forwarding_closure(
     }
 }
 
-pub(crate) fn get_signature_info(signature: &venial::Function, has_gd_self: bool) -> SignatureInfo {
+/// Maps each usage of `Self` to the struct it's referencing,
+/// since `Self` can't be used inside nested functions.
+fn map_self_to_class_name<T: FromIterator<TokenTree>>(
+    tokens: impl IntoIterator<Item = TokenTree>,
+    class_name: &Ident,
+) -> T {
```

Comment:
I would probably use a `where` close with two named type parameters `In` and `Out`, for symmetry reasons.